main.go

```diff
@@ -401,49 +401,146 @@ func connectAction(ctx *cli.Context) error {
 	return nil
 }
 
-// disconnectAction tries to stop rhscd service, disconnect from Red Hat Insights and finally
-// it unregister system from Red Hat Subscription Management
+// setupFormatOption ensures the user has supplied a correct `--format` flag
+// and set values in uiSettings, when JSON format is used.
+func setupFormatOption(ctx *cli.Context) error {
+	// This is run after the `app.Before()` has been run,
+	// the uiSettings is already set up for us to modify.
+	format := ctx.String("format")
+	switch format {
+	case "":
+		return nil
+	case "json":
+		uiSettings.isMachineReadable = true
+		uiSettings.isRich = false
+		return nil
+	default:
+		err := fmt.Errorf(
+			"unsupported format: %s (supported formats: %s)",
+			format,
+			`"json"`,
+		)
+		return cli.Exit(err, 1)
+	}
+}
+
+// DisconnectResult is structure holding information about result of
+// disconnect command. The result could be printed in machine-readable format.
+type DisconnectResult struct {
+	Hostname                  string `json:"hostname"`
+	HostnameError             string `json:"hostname_error,omitempty"`
+	UID                       int    `json:"uid"`
+	UIDError                  string `json:"uid_error,omitempty"`
+	RHSMDisconnected          bool   `json:"rhsm_disconnected"`
+	RHSMDisconnectedError     string `json:"rhsm_disconnect_error,omitempty"`
+	InsightsDisconnected      bool   `json:"insights_disconnected"`
+	InsightsDisconnectedError string `json:"insights_disconnected_error,omitempty"`
+	YggdrasilStopped          bool   `json:"yggdrasil_stopped"`
+	YggdrasilStoppedError     string `json:"yggdrasil_stopped_error,omitempty"`
+	format                    string
+}
+
+// Error implement error interface for structure DisconnectResult
+// It is used for printing DisconnectResult by cli.Exit()
+func (disconnectResult DisconnectResult) Error() string {
+	var result string
+	switch disconnectResult.format {
+	case "json":
+		data, err := json.MarshalIndent(disconnectResult, "", "    ")
+		if err != nil {
+			return err.Error()
+		}
+		result = string(data)
+	default:
+		result = "error: unsupported document format: " + disconnectResult.format
+	}
+	return result
+}
+
+// beforeDisconnectAction ensures the used has supplied a correct `--format` flag
+func beforeDisconnectAction(ctx *cli.Context) error {
+	return setupFormatOption(ctx)
+}
+
+// rhcPrintf is method for printing human-readable output. It suppresses output, when
+// machine-readable format is used.
+func rhcPrintf(format string, a ...interface{}) {
```

Comment:
The name of this function doesn't indicate its purpose clearly to me. Could we
call it something like `interactivePrintf`?